File: CSA2/sourse files/tree.cpp
//------------------------------------------------------------------------------
// tree.cpp - содержит реализацию класса дерева
//------------------------------------------------------------------------------

#include "tree.h"

Tree::Tree(std::string name, long treeAge)
	:Plant(name)
{
	age = treeAge;
}

//------------------------------------------------------------------------------
// Вывод параметров дерева в виде строки
std::string Tree::ToString() {
	return "It is Tree: name = " + Name
		+ ", age = " + std::to_string(age)
		+ ". Part of Vowels = " + std::to_string(GetPartOfVowels());
};

File: CSA3/sourse files/tree.py
from plant import *


# ----------------------------------------------
class Tree(Plant):
    age = 0

    def __init__(self, name, age):
        super().__init__(name)
        self.age = age

    # Возвращает строковое представление растения
    def toString(self):
        return "It is Tree: name = {}, age = {}. Part of Vowels = {}".format(self.name, self.age,
                                                                             self.getPartOfVowels())
